### PR TITLE
Allow bypassing mime checks for 3rdparty checks. on request by @mbabker

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Allow bypassing mime checks for 3rdparty checks. on request by @mbabker

### Testing Instructions

Try to upload a image or PDF file using com_media.

In the CMS itself there is no call to that bypass.

### Expected result

mime checks are bypassed now (similiar to the raw filter in JInput)

### Actual result

no way to bypass the mime checks as developer

### CODE REVIEW

File: libraries/cms/helper/media.php [hunk 1]
```diff
@@ -131,14 +131,15 @@ private function checkMimeType($mime, $component = 'com_media')
...
+	 * @param   array    $file       File information
+	 * @param   string   $component  The option name for the component storing the parameters
+	 * @param   boolean  $checkMime  With this option you can disable all mime checks
 	 *
 	 * @return  boolean
 	 *
 	 * @since   3.2
 	 */
-	public function canUpload($file, $component = 'com_media')
+	public function canUpload($file, $component = 'com_media', $checkMime = ture)
```
Comment (mbabker) [inline libraries/cms/helper/media.php]: Typo, should be `true`

**Review by mbabker — DISMISSED** [review]

File: libraries/cms/helper/media.php (hunk 1 above)
Comment (zero-24) [reply libraries/cms/helper/media.php]: fixed thanks.

File: libraries/cms/helper/media.php [hunk 2]
```diff
@@ -131,14 +131,15 @@ private function checkMimeType($mime, $component = 'com_media')
 	/**
 	 * Checks if the file can be uploaded
 	 *
-	 * @param   array   $file       File information
-	 * @param   string  $component  The option name for the component storing the parameters
+	 * @param   array    $file       File information
+	 * @param   string   $component  The option name for the component storing the parameters
+	 * @param   boolean  $checkMime  With this option you can disable all mime checks
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change suggestion: `If false, disable all mime checks`

File: libraries/cms/helper/media.php [hunk 3]
```diff
@@ -215,28 +216,32 @@ public function canUpload($file, $component = 'com_media')
...
+					if ($checkMime)
 					{
-						$result = $this->checkMimeType($mime, $component);
+						// Get the mime type this is an image file
+						$mime = $this->getMimeType($file['tmp_name'], true);
 
-						// If the mime type is not allowed we don't upload it and show the mime code error to the user
-						if ($result === false)
+						// Did we get anything useful?
+						if ($mime != false)
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change to `!==`?

File: libraries/cms/helper/media.php [hunk 4]
```diff
@@ -248,29 +253,33 @@ public function canUpload($file, $component = 'com_media')
...
-				if ($mime != false)
+				// Bypass all mime checks...
+				if ($checkMime)
 				{
-					$result = $this->checkMimeType($mime, $component);
+					// Get the mime type this is not an image file
+					$mime = $this->getMimeType($file['tmp_name'], false);
+
+					// Did we get anything useful?
+					if ($mime != false)
```
Comment (Quy) [inline libraries/cms/helper/media.php]: Change to `!==`?

**Comment by zero-24** [comment]
done @Quy Thanks.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 05adbad0385bbad3c0ac995dfb6df52626b89ebd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16244">issues.joomla.org/tracker/joomla-cms/16244</a>.</sub>

**Comment by zero-24** [comment]
@mbabker it looks like no one is interested in that change? So should we just close it?

**Comment by zero-24** [comment]
Closing as there was a broken merge conflict fix.

**Pull request closed without merging**